test/integration/cloudbuild_repo_connection_gitlab/cloudbuild_repo_connection_gitlab_test.go

```diff
@@ -0,0 +1,165 @@
+		NamespaceID: gitlab.Ptr(gl.namespace),
+		// Required otherwise Cloud Build errors on creating the connection
+		InitializeWithReadme: gitlab.Ptr(true),
+	}
+	proj, _, err := gl.client.Projects.CreateProject(opts)
+	if err != nil {
+		gl.t.Fatal(err.Error())
+	}
+	gl.project = proj
+}
+
+func (gl *GitLabClient) AddFileToProject(file []byte) {
+	opts := &gitlab.CreateFileOptions{
+		Branch:        gitlab.Ptr("main"),
+		CommitMessage: gitlab.Ptr("Initial config commit"),
+		Content:       gitlab.Ptr(string(file)),
+	}
+	_, _, err := gl.client.RepositoryFiles.CreateFile(gl.ProjectName(), "main.tf", opts)
+	if err != nil {
+		gl.t.Fatal(err.Error())
+	}
+}
+
+func (gl *GitLabClient) DeleteProject() {
+	resp, err := gl.client.Projects.DeleteProject(gl.ProjectName())
+	if resp.StatusCode != 404 && err != nil {
+		gl.t.Errorf("error deleting project with status %s and error %s", resp.Status, err.Error())
+	}
+	gl.project = nil
+}
+
+func TestCloudBuildRepoConnectionGitLab(t *testing.T) {
+	repoName := fmt.Sprintf("conn-gl-%s", utils.GetRandomStringFromSetup(t))
+	gitlabPAT := cftutils.ValFromEnv(t, "IM_GITLAB_PAT")
+	owner := "infrastructure-manager"
+
+	client := NewGitLabClient(t, gitlabPAT, owner, repoName)
+	proj := client.GetProject()
+	if proj == nil {
+		client.CreateProject()
+	}
+
+	resourcesLocation := "us-central1"
+	vars := map[string]interface{}{
+		"gitlab_read_authorizer_credential": gitlabPAT,
+		"gitlab_authorizer_credential":      gitlabPAT,
+		"repository_name":                   repoName,
+		"repository_url":                    client.project.HTTPURLToRepo,
+	}
+	bpt := tft.NewTFBlueprintTest(t, tft.WithVars(vars))
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		bpt.DefaultVerify(assert)
+
+		t.Cleanup(func() {
+			// Delete the repository if we hit a failed state
+			if t.Failed() {
+				client.DeleteProject()
+			}
+		})
+
+		// validate if repository was created using the connection
+		projectId := bpt.GetTFSetupStringOutput("project_id")
+		connectionId := bpt.GetStringOutput("cloud_build_repositories_2nd_gen_connection")
+
+		connectionIdRegexPattern := `^projects/[^/]+/locations/[^/]+/connections/[^/]+$`
+		re := regexp.MustCompile(connectionIdRegexPattern)
+		assert.True(re.MatchString(connectionId), "Connection ID should be in format projects/{{project}}/locations/{{location}}/connections/{{name}}")
+
+		connectionSlice := strings.Split(connectionId, "/")
+		// Extract the project, location and connection name from the connection_id
+		connectionProjectId := connectionSlice[1]
+		connectionLocation := connectionSlice[3]
+		connectionName := connectionSlice[len(connectionSlice)-1]
+
+		// Assert that the resource was created in the specified project and region
+		assert.Equal(projectId, connectionProjectId, "Connection project id should be the same as input project id.")
+		assert.Equal(resourcesLocation, connectionLocation, "Connection location should be 'us-central'.")
```

Comment:
`us-central1`? Perhaps better to use `resourcesLocation` in `fmt.Sprintf`.

```suggestion
		assert.Equal(resourcesLocation, connectionLocation, "Connection location should be 'us-central'.")
```